src/Analysis.cpp

```diff
@@ -580,11 +592,64 @@ void CodeAnalyzer::doAnalysis(Function &F, std::string optLoadFileName) {
     }
 }
 
+// assumes utilFunctionTester is actually an extended class and not UtilFunctionTester
+// (e.g., an instance of VariableTester)
+void runUtilFunctionTester(UtilFunctionTester* utilFunctionTester, const std::string& functionName) {
+    errs() << "RUNNING UTIL FUNCTION TEST: " << functionName << "\n";
+    if (!utilFunctionTester->runTest()) {
+        errs() << "UTIL FUNCTION TEST FAIL: " << functionName << "\n";
+        std::exit(EXIT_FAILURE);
+    }
+
+    errs() << "UTIL FUNCTION TEST PASS: " << functionName << "\n\n";
+}
+
 void CodeAnalyzer::onEnd() {
     if (BUILD_PROGRAM_LINES_BRANCH_INFO) {
         programLinesBranchesInfo.generate(cFileName, true);
     }
 
+    // rlc_dataflow function testers
```

Comment:
This is more of a higher-level design question, but it worries me a bit that these (unit) tests are triggered by `onEnd()`. We will eventually want to run the analyzer without running the tests (e.g., to analyze a kernel module), so having the tests be triggered by the analyzer run ending is probably not the design that we want long-term. At a minimum, I think we should write a helper function (e.g., `runAllTests()` or something like that) that triggers all of the tests and call it here, so that there's only one place that needs to be changed when we eventually want to put these tests behind a flag.